### PR TITLE
[CMR] M3-4346: Table styles for StackScripts

### PR DESCRIPTION
**Pull request by tiffwong**
## Description
Applied the CMR table styles to StackScripts and added a column to display whether it is public or private. I currently have the column labeled as "Status" since I couldn't think of a better way to describe it. The "Status" column is hidden for tablet which is followed by hiding "Last Revision" and "Compatible Images" for mobile.

## Type of Change
- Non breaking change ('update', 'change')

## Note to Reviewers
The StackScripts component is shared between the landing page, the Linode Create flow and the Linode Rebuild flow. Please check all three areas for any regression.

### CODE REVIEW

File: packages/manager/src/features/StackScripts/Partials/StackScriptTableHead_CMR.tsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+import * as React from 'react';
+import Hidden from 'src/components/core/Hidden';
+import { makeStyles, Theme } from 'src/components/core/styles';
+import TableHead from 'src/components/core/TableHead';
+import TableRow from 'src/components/TableRow/TableRow_CMR';
+import TableCell from 'src/components/TableCell/TableCell_CMR';
+import TableSortCell from 'src/components/TableSortCell/TableSortCell_CMR';
+
+// @todo: the CMR styles for TableSortCell aren't being applied and I can't
+// figure out why so I had to use '!important' to overwrite them
```
Comment (Jskobos) [inline packages/manager/src/features/StackScripts/Partials/StackScriptTableHead_CMR.tsx:11]: Still seems a little off:

Screen Shot 2020-08-17 at 8 30 19 AM

File: packages/manager/src/features/linodes/LinodesDetail/LinodeRebuild/LinodeRebuildDialog.tsx [hunk 2]
```diff
@@ -17,8 +17,14 @@ const useStyles = makeStyles((theme: Theme) => ({
...
-      '& div': {
-        padding: 0
+      '& .MuiPaper-root': {
+        padding: 0,
+        '& > div': {
+          padding: 0
+        },
+        '& .MuiTableCell-head': {
+          top: theme.spacing(11)
+        }
```
Comment (tiffwong) [inline packages/manager/src/features/linodes/LinodesDetail/LinodeRebuild/LinodeRebuildDialog.tsx:27]: This is to fix the spacing the sticky header used in the StackScriptPanel

File: packages/manager/src/features/StackScripts/Partials/StackScriptTableHead_CMR.tsx (hunk 1 above)
Comment (tiffwong) [reply packages/manager/src/features/StackScripts/Partials/StackScriptTableHead_CMR.tsx:11]: I updated it to use the Landing Header component that you recommended and also removed the empty column in the Linode Create flow to make it less weird, lmk what you think!

**Comment by tiffwong** [comment]
> https://www.loom.com/share/ff786c6ae638482d9c38bc1b74f2a4e3

@DevDW is that still happening for you now?

**Comment by DevDW** [comment]
> > https://www.loom.com/share/ff786c6ae638482d9c38bc1b74f2a4e3
> 
> @DevDW is that still happening for you now?

No, the slowness is gone and I'm not observing the constant StackScript requests now. 👍🏾 

The only other thing I noticed during a quick look-through is that the StackScript table column headers aren't bold as they are for other table column headers like Linodes.

**Comment by tiffwong** [comment]
@Jskobos @WilkinsKa1 I added the changes you requested, lmk if there's anything else missing

**Review by Jskobos — APPROVED** [review]
So sorry about this, didn't see the re-requested review. Everything looks good. There's some negative margin or something with the Paper that makes the effective padding above the Create a StackScript button smaller than it is elsewhere (it's 8px everywhere but half of that is above the paper here). This isn't blocking.

I'd personally like a little less whitespace between the header and tabs, but maybe that's there for an error? and in any case just my preference.

**Pull request merged**